### ISSUE TITLE
Electronic Scoring

### ISSUE THREAD

**Issue by moriarty** [RoboCupAtHome/RuleBook#354]
A while ago, I used LimeSurvey @LimeSurvey for a project, and it was still new and a little buggy but worked... When someone mentioned the scoring should be automated, it came to mind.

It looks like it's now the number 1 FOSS online survey software.

Might be worth looking into for automating the scoring.

It's server based and done in php- so the judges would need access to a hidden wifi, and could be given access codes to score the test. But any smartphone/tablet/laptop could then be used to score.

**Comment by kyordhel** [comment]
(as I said in #355)

This requires a significant amount of work. The OC will accept volunteers to develop a web app for scoring the next Rulebook (if suitable) once the final version has been published and only if this is published 2 months before the competition.

As side note, after refereeing OPL alone in a couple of tests, I'm pretty sure I won't do that in any app/interface. There are many notes to take during the tests, and nothing outperforms the efficiency of pen and paper for this purpose.

**Comment by LoyVanBeek** [comment]
I started developing a small webapp to enter scoresheets with before the German Open, but indeed nothing trumps pen and paper for this task, so I stopped. Especially adding short notes when I'm unsure to give eg. full points or a partial score, that sort of thing.

**Comment by kyordhel** [comment]
Sometimes robots slightly collide while grasping objects, so you need to write down why you aren't scoring even though the robot "accomplished" the task (kinda should stop it here but let's continue the show).

In GPSR and EEGPSR it turns out much more critical, specially when robots don't get the command correctly but close enough (eg [...] and follow X from the Teepee to the Teepee). Again there is no way to score but you need to write down why and what was happening. Something as simpler as writing the command given to the robot can become time consuming if you also need to deal with the autocorrect of an app.

**Comment by LoyVanBeek** [comment]
I really want to have digital scoresheet, but with this in mind, I now do understand why there is such a high administrative burden in eg. healthcare. 

Can't we just enter the scoresheets by voice input as well, at least the notes :smiley: ?

**Comment by kyordhel** [comment]
We can use Watson for scoring in that case... or any robot for that matter.

**Comment by kyordhel** [comment]
Electronic scoring is **the** way to go, but only after the rulebook's test have been designed accordingly (to score Boolean) and in such a way that compiling the rulebook will generate both: printable and electronic Scoresheets (pdf forms is not an option).

**Comment by moriarty** [comment]
I'm on holiday but it's raining, so I'll give a quick update... 

I looked into using limesurvey, which we would then need to create surveys which are actually score sheets... This is a very manual process 👎  but then when done, it automatically tallies and collects the scores, we even get live data as the scores are entered 👍 ... but of course, this requires wifi 👎 ... The referees would get codes to access the survey from their phones/tablets/laptops. The lime wire survey runs on anything, I was testing via the docker image.

But, the setup led me to looking into a different direction. We have the score sheets in Latex, so I looked for LaTeX -> HTML options. Thinking that it would be best if we only need to maintain the score sheets in one location, and we use the existing location.

And I found this **\LaTeX** package: http://www.math.uakron.edu/~dpstory/eqexam.html it's a little old, but if it works, it might be a good middle step.

**Comment by justinhart** [comment]
I actually really like the idea of electronic scoresheets and perhaps a
computer monitor that teams can look at with a current electronic
scoreboard. It would make it all feel somehow more official and
television-ready and such.

What if we just hacked something together on our own? It might be easier to
customize. The only part of all of this that I'm unfamiliar with packages
to manage user accounts on a web app. But, if the entire TC registered with
a quick website that we put up, they could punch in their scores as we go
and we could make team-lead sign-offs also an online thing. It would
streamline the whole thing and reduce the amount of running around during
the competition, as well as give us a cool live scoreboard.

I think that a LAMP website would be a lot easier than hacking together
something out of LaTeX to print out paper scoresheets. We could just do
away with paper altogether and make our lives a bit easier.

**Comment by LoyVanBeek** [comment]
I whipped womething up in to run up to the GermanOpen: https://github.com/LoyVanBeek/ScoreCruncher. It parses the LaTeX files of the scoresheets and servers a webpage. 

That is entirely possible, but I'm mostly worried about the ergonomics of doing so, because often there are remarks etc. Could be another field in the form though.

**Comment by justinhart** [comment]
I like the idea of just adding a comment form onto the page. I get that the
current process isn't terribly terribly laborious, but sticking everything
into a website could streamline things on-site, which will already be
chaos. The bigger side of this, for me, is that I think that a live
scoreboard would be really really cool.

**Comment by LoyVanBeek** [comment]
Another feature (not sure if I implemented it that way anymore) is to be able to give partial scores, not just a checkbox always yielding a fixed amount of points. 
Instead, perhaps have a button that inserts the default score in a textfield but also be able to enter a custom score.

**Comment by kyordhel** [comment]
Considering that no major changes will be done to the rulebook, I stick with pen and paper. So far my handwriting is 30 times faster than my skills typing in an on-screen keyboard

**Comment by LoyVanBeek** [comment]
Indicates poor UX on the forms you'd use, so that needs to be good enough, also on eg. your phone and laptop. 

Doing this electronically makes us finish and publish the scores 30x faster in return.

**Comment by justinhart** [comment]
I just think that the electronic scoreboard would be fun.

**Comment by kyordhel** [comment]
Let's try to have this by 2020

**Comment by kyordhel** [comment]
Related to #197

**Comment by airglow** [comment]
Under https://github.com/airglow/rc_home_digial_scoresheet, i created a small script which takes the score sheet tex files as input for iteratively grading. In the end a report is generated. Note this is just an idea. Some score sheets should be restructured for making it more optimal. I.e. for Storing Groceries the correct placements are on top, meaning the script would just ask for the correct placements first and then the incorrect placements. Alternating those makes more sense for scoring. 

If you like  something like this I could progress. Adding a live score or a text on what currently will be graded should be straight forward. May  in combination with #389.

**Comment by moriarty** [comment]
As much as I would like this, I think with the upcoming changes to the RuleBook this will become unlikely 🙁

**Comment by johaq** [comment]
Closed. I think for scoring inside the arena during a test, paper is still the way to go. You need to quickly make changes and annotations if necessary.
